### PR TITLE
feat(presets): add eslint monorepo

### PR DESCRIPTION
**Pull request by fuxingloh**
## Changes

ESLint new flat config ships a new package with scope `@eslint`. 

- https://github.com/eslint/eslint/tree/main/packages/js
- https://eslint.org/docs/latest/use/configure/configuration-files-new#using-predefined-configurations

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/monorepo.ts [hunk 1]
```diff
@@ -265,6 +265,7 @@ const repoGroups = {
   'ember-decorators': 'https://github.com/ember-decorators/ember-decorators',
   emojibase: 'https://github.com/milesj/emojibase',
   emotion: 'https://github.com/emotion-js/emotion',
+  eslint: 'https://github.com/eslint/eslint',
```
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:268]: Do any of our existing rules already group eslint packages and would conflict with this? If so then we need to decide which changes to make

Comment (viceice) [reply lib/config/presets/internal/monorepo.ts:268]: we've some linter grouping rules

Comment (rarkins) [reply lib/config/presets/internal/monorepo.ts:268]: Those are opt-in I think

Comment (rarkins) [reply lib/config/presets/internal/monorepo.ts:268]: @fuxingloh how many packages are published from github.com/eslint/eslint other than eslint itself?

Comment (rarkins) [reply lib/config/presets/internal/monorepo.ts:268]: Seems not many: https://github.com/eslint/eslint/tree/main/packages

**Review by rarkins — APPROVED** [review]

**Pull request merged**